src/osd/PGLog.h

```diff
@@ -713,9 +720,10 @@ struct PGLog : DoutPrefixProvider {
   void roll_forward_to(
     eversion_t roll_forward_to,
     LogEntryHandler *h) {
-    log.roll_forward_to(
-      roll_forward_to,
-      h);
+    if (log.roll_forward_to(
+	  roll_forward_to,
+	  h))
+      touched_log = false;
```

Comment:
This bool is convenient, but it really isn't the right way dirty the log.  Not sure of a better way to go.

Reply:
we could modify behavior of PGLog::is_dirty(), will it be a better way?